File: services/dashboard-ui/client/lib/ctl-api/installs/get-install-stack.ts
import { api } from '@/lib/api'
import type { TInstallStack } from '@/types'

export const getInstallStack = ({
  installId,
  orgId,
}: {
  installId: string
  orgId: string
}) =>
  api<TInstallStack>({
    path: `installs/${installId}/stack`,
    orgId,
  })

File: services/ctl-api/internal/app/installs/worker/activities/get_install_stack.go
package activities

import (
	"context"

	"github.com/nuonco/nuon/services/ctl-api/internal/app"
	"github.com/nuonco/nuon/services/ctl-api/internal/pkg/db/generics"
)

type GetInstallStackRequest struct {
	InstallID string `json:"id" validate:"required"`
}

// @temporal-gen-v2 activity
// @by-field InstallID
func (a *Activities) GetInstallStack(ctx context.Context, req GetInstallStackRequest) (*app.InstallStack, error) {
	var stack app.InstallStack

	res := a.db.WithContext(ctx).
		Where(app.InstallStack{
			InstallID: req.InstallID,
		}).
		Preload("InstallStackOutputs").
		First(&stack)
	if res.Error != nil {
		return nil, generics.TemporalGormError(res.Error, "unable to get install stack")
	}

	return &stack, nil
}
